### PR TITLE
HBASE-27680 Bump hbase, hbase-thirdparty, hadoop and spark for hbase-…

### PR DESCRIPTION
**Pull request by NihalJain**
…connectors

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -138,14 +138,14 @@
...
+    <hbase.version>2.4.16</hbase.version>
     <exec.maven.version>1.6.0</exec.maven.version>
     <audience-annotations.version>0.5.0</audience-annotations.version>
     <junit.version>4.12</junit.version>
     <mockito-all.version>1.8.5</mockito-all.version>
-    <hbase-thirdparty.version>4.0.1</hbase-thirdparty.version>
-    <hadoop-two.version>2.8.5</hadoop-two.version>
-    <hadoop-three.version>3.2.0</hadoop-three.version>
+    <hbase-thirdparty.version>4.1.4</hbase-thirdparty.version>
+    <hadoop-two.version>2.10.0</hadoop-two.version>
```
Comment (NihalJain) [inline pom.xml:147]: Should we bump to hadoop 2.10.2?

File: spark/pom.xml [hunk 2]
```diff
@@ -45,10 +45,10 @@
   <properties>
     <protobuf.plugin.version>0.6.1</protobuf.plugin.version>
     <jackson.version>2.12.5</jackson.version>
-    <spark.version>3.1.2</spark.version>
+    <spark.version>3.3.2</spark.version>
```
Comment (NihalJain) [inline spark/pom.xml:48]: Should we bump to spark 3.2.3?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 46s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   1m 30s |  master passed  |
| +1 :green_heart: |  compile  |   0m 56s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m  3s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   1m  1s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   8m 44s |  root in the patch passed.  |
|  |   |  18m 55s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-113/1/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-connectors/pull/113 |
| Optional Tests | dupname javac javadoc unit xml compile |
| uname | Linux 9f1f3f1ae364 5.4.0-1092-aws #100~18.04.2-Ubuntu SMP Tue Nov 29 08:39:52 UTC 2022 x86_64 GNU/Linux |
| Build tool | hb_maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / bb2902f |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-113/1/testReport/ |
| Max. process+thread count | 963 (vs. ulimit of 12500) |
| modules | C: spark . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-Connectors-PreCommit/job/PR-113/1/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by NihalJain** [comment]
This patch changes following:
<div class="table-wrap" style="margin: 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Component | Old | New
-- | -- | --
HBase | 2.4.9 | 2.4.16
HBase Thirdparty | 4.0.1 | 4.1.4
Hadoop 2 | 2.8.5 | 2.10.0
Hadoop 3 | 3.2.0 | 3.2.4
Spark | 3.1.2 | 3.3.2
Scala | 2.12 | 2.12 (no change)
Scala Binary | 2.12.10 | 2.12.15

</div><p style="margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Ran following and ensured build SUCCESS:</p><ul style="margin: 10px 0px 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li>mvn  clean install</li><li>mvn  clean install -Dhadoop.profile=2.0</li></ul><p style="margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Built with following for previous version build and ensured assembly tar ball looks same:</p><ul style="margin: 10px 0px 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li>mvn  clean install -Dspark.version=3.1.2 -Dscala.binary.version=2.12 -Dscala.version=2.12.10 -Dhadoop-three.version=3.2.0 -Dhadoop-two.version=2.8.5 -Dhbase-thirdparty.version=4.0.1 -Dhbase.version=2.4.9</li><li>mvn clean install -Dhadoop.profile=2.0 -Dspark.version=3.1.2 -Dscala.binary.version=2.12 -Dscala.version=2.12.10 -Dhadoop-three.version=3.2.0 -Dhadoop-two.version=2.8.5 -Dhbase-thirdparty.version=4.0.1 -Dhbase.version=2.4.9</li></ul><br class="Apple-interchange-newline">

**Review by meszibalu — APPROVED** [review]

**Pull request merged**